### ISSUE TITLE
Linking fails with clang-cl compiler Release build 

### ISSUE THREAD

**Issue by qno** [free-audio/clap-helpers#61]
https://github.com/free-audio/clap-helpers/blob/ea88647873d22f8181940729507fee3b2a3db5bf/include/clap/helpers/host-proxy.hxx#L46

If https://github.com/free-audio/clap-saw-demo-imgui is compiled with [clang-cl](https://clang.llvm.org/docs/UsersManual.html#clang-cl) as **Release** build, the following linker error occurs `lld-link: error: undefined symbol: public: void __cdecl clap::helpers::HostProxy<1, 2>::getExtension<struct clap_host_timer_support>(struct clap_host_timer_support const *&, char const *)`

It fails if compiled on Windows or Crosscompiled on Linux. It links only when build as RelWithDebInfo or Debug or MSVC toolchain is used.

```
====================[ Build | clap-saw-demo-imgui | Release-Visual Studio clang-cl ]====
"C:\Program Files (x86)\JetBrains\CLion 2023.3.4\bin\cmake\win\x64\bin\cmake.exe" --build C:\Users\Silvio\Desktop\test\clap-saw-demo-imgui\build --target clap-saw-demo-imgui --config Release -j 14
[1/1] Linking CXX shared module Release\clap-saw-demo-imgui.clap
FAILED: Release/clap-saw-demo-imgui.clap 
cmd.exe /C "cd . && "C:\Program Files (x86)\JetBrains\CLion 2023.3.4\bin\cmake\win\x64\bin\cmake.exe" -E vs_link_dll --intdir=CMakeFiles\clap-saw-demo-imgui.dir\Release --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100220~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100220~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2022\BUILDT~1\VC\Tools\Llvm\x64\bin\lld-link.exe /nologo CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo.cpp.obj CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo-editor.cpp.obj CMakeFiles\clap-saw-demo-imgui.dir\Release\src\saw-voice.cpp.obj CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo-pluginentry.cpp.obj  /out:Release\clap-saw-demo-imgui.clap /implib:Release\clap-saw-demo-imgui.lib /pdb:Release\clap-saw-demo-imgui.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO  libs\clap-imgui-support\Release\imgui-clap-support.lib  d3d12.lib  dxgi.lib  d3dcompiler.lib  dxguid.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK: command "C:\PROGRA~2\MICROS~2\2022\BUILDT~1\VC\Tools\Llvm\x64\bin\lld-link.exe /nologo CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo.cpp.obj CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo-editor.cpp.obj CMakeFiles\clap-saw-demo-imgui.dir\Release\src\saw-voice.cpp.obj CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo-pluginentry.cpp.obj /out:Release\clap-saw-demo-imgui.clap /implib:Release\clap-saw-demo-imgui.lib /pdb:Release\clap-saw-demo-imgui.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO libs\clap-imgui-support\Release\imgui-clap-support.lib d3d12.lib dxgi.lib d3dcompiler.lib dxguid.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST:EMBED,ID=2" failed (exit code 1) with the following output:
lld-link: error: undefined symbol: public: void __cdecl clap::helpers::HostProxy<1, 2>::getExtension<struct clap_host_timer_support>(struct clap_host_timer_support const *&, char const *)
>>> referenced by CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo-editor.cpp.obj:(protected: virtual bool __cdecl sst::clap_saw_demo::ClapSawDemo::guiCreate(char const *, bool))
>>> referenced by CMakeFiles\clap-saw-demo-imgui.dir\Release\src\clap-saw-demo-editor.cpp.obj:(protected: virtual void __cdecl sst::clap_saw_demo::ClapSawDemo::guiDestroy(void))
ninja: build stopped: subcommand failed.
```

**Comment by baconpaul** [comment]
just to manage expectations, the clap saw demo imgui doesn't work very well. The original with vstgui works better.

But this is a weird error. Looks like it is getting optimized out but then a symbol is being ejected.  This is almost definitely because the editor object includes the host. h but not the host.hxx so the implicit symbol is getting knocked out.

I have no way to test this but I wonder if including at this like

https://github.com/free-audio/clap-saw-demo-imgui/blob/96da3ab3ed9bc9cc26eecae691db8104bd2d2f79/src/clap-saw-demo-editor.cpp#L8

```
#include <clap/helpers/host-proxy.hxx>
```

would fix it. Since it is a double template it needs to be visible to the person t instantiate it and only the editor does in the imgur project.

Alternately ask for the timer extension in the non-editor file and that will resolve it

Can you try either of those?

**Comment by qno** [comment]
The `#include <clap/helpers/host-proxy.hxx>` works.

I was just using this app as kind of more complex hello-world example for testing crosscompiling native Windows binaries with clang-cl and CMake under Linux and then that issue occured. Then I checked on Windows and same error happend as well. 

I just opened the issue to check if this could be an issue with the clap-helper or if it is just not used correctly in the demo app. Thank you for looking into this.
